src/reducers/ui.js

```diff
@@ -85,6 +85,20 @@ function update(
     }
 
     case constants.TOGGLE_FILE_SEARCH_MODIFIER: {
+      const actionVal = !state.getIn(["fileSearchModifiers", action.modifier]);
+
+      if (action.modifier == "caseSensitive") {
+        prefs.fileSearchCaseSensitive = actionVal;
```

Comment:
I am not very familiar with the issue, but should we also set the state in these cases?